### PR TITLE
Fix the New York working families tax credit logic to properly account for filers with both older and younger dependents

### LINKED ISSUE #5326: Refactor the New York Working families tax credit
Currently the reduction does not apply fully for the EITC along with interactions between older and younger children

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #5326

### CODE REVIEW

File: policyengine_us/parameters/gov/contrib/states/ny/wftc/amount/max.yaml [hunk 1]
```diff
@@ -13,9 +13,8 @@ metadata:
...
 values:
-  2024-01-01: 0
-# 2024: 550
-# 2025: 800
-# 2026: 1,000
-# 2027: 1,200
-# 2028: 1,600
+  2025-01-01: 550
+  2026-01-01: 800
+  2027-01-01: 1000
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/contrib/states/ny/wftc/amount/max.yaml:18]:
```suggestion
  2027-01-01: 1_000
```
And elsewhere

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Is this a refactor or does it fix/amend logic?

**Comment by PavelMakarchuk** [comment]
> Is this a refactor or does it fix/amend logic?

We amend the logic to properly account for families with older and younger children as well as the treatment of the parents with children - renamed

**Comment by PavelMakarchuk** [comment]
> > Is this a refactor or does it fix/amend logic?
> 
> We amend the logic to properly account for families with older and younger children as well as the treatment of the parents with children - renamed

Also refactored all of the tests as some were outdated

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**